site/src/components/Modal/CreateSchemaModal.tsx

```diff
@@ -0,0 +1,125 @@
+import { Box, Typography } from "@mui/material";
+import { FC, useState, FormEvent, useCallback } from "react";
+import { useRouter } from "next/router";
+import { Modal } from "./Modal";
+import { apiClient } from "../../lib/apiClient";
+import { Button } from "../Button";
+import { useUser } from "../../context/UserContext";
+import { TextField } from "../TextField";
+
+type CreateSchemaModalProps = {
+  open: boolean;
+  onClose: () => void;
+};
+
+export const CreateSchemaModal: FC<CreateSchemaModalProps> = ({
+  open,
+  onClose,
+}) => {
+  const [newSchemaTitle, setNewSchemaTitle] = useState("");
+  const [loading, setLoading] = useState(false);
+  const [error, setError] = useState("");
```

Comment:
I think this should be `undefined` by default, and support `ReactNode` incase we want to introduce formatting in the message. 
```suggestion
  const [error, setError] = useState<ReactNode>();
```

Reply:
I don't think there's a need to support that until we eventually introduce formatting the message. Will update for consistency

Reply:
Fixed here => https://github.com/blockprotocol/blockprotocol/pull/223/commits/cc8b0a31aafe03be4c128127b8c83292084ea4d9